Rohith + Raj - multiple project can be added, create project button implemented. AI chat history linked to current project. AI RAG linked to current project.

rohith-chaps88 opened the pull request:
# Root Cause
<!--- Please describe why this change is required. What's the motive? --> This change is to connect the front end to the backend, where the create project button can access the backend database where project are held, and the buttoon allows to add projects to the database. Furthermore, frontend was developed to show the added projects in the NavBar.

AI Chat history is now linked to current project. AI knowledge now linked ot currenlty selected project

# Solution 
<!--- Please describe your code changes in detail for developers. --> 
Backend (FastAPI + SQLite): adds /projects + /projects/{id}/transcriptions APIs backed by our existing database_models.py (sqlite3 helper).

Frontend : connects New Project modal to POST to the API and refresh the NavBar project bar on success.
- New ProjectContext.jsx introduced. Front end refactored to utlise new context.

# How Has This Been Tested? 
<!--- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration --> 

# Checklist: 
- [ ] I have linked issue to this Pull Request 
- [ ] I have performed a self-review of my code 
- [ ] I have commented my code, particularly in hard-to-understand areas 
- [ ] I have made corresponding changes to the documentation 
- [ ] I have added tests that prove my fix is effective or that my feature works 
- [ ] New and existing unit tests pass locally with my changes

rohith-chaps88 committed:
api endpoints were created, as well as front end new project button to connect to backend.

rohith-chaps88 committed:
Merge remote-tracking branch 'origin/develop' into rohith-database_multiple_transactions

rohith-chaps88 committed:
create new project button connected to backend

Gunnraj-D commented on server/app/main.py:
```diff
@@ -150,3 +173,130 @@ async def download_transcription(final_output: str = Form(...), filename: str =
+        raise HTTPException(status_code=404, detail="Project not found")
+
+    rows = transcripts_store.get_all_project_transcriptions(project_id)
+    return [_transcription_meta_row_to_dict(r) for r in rows]
```
Project and transcription helper functions in main.py should probably appear in the "Project & Transcription routes" section, preferably underneath the endpoints.

Gunnraj-D commented on server/app/main.py:
```diff
@@ -44,6 +49,13 @@ async def lifespan(app: FastAPI):
+DB_PATH = str((Path(__file__).resolve().parent / "qualAI.db"))
+projects_store = db.Project(DB_PATH)
+transcripts_store = db.Transcription(DB_PATH)
+
```
couldn't this happen in the lifespan function?

Gunnraj-D reviewed (changes requested):
- AIAssistant component should send the current project as well as the prompt for the GENERATE endpoint, as that determines which project the AI will look through
- The same AIAssistant text history should not be shown for each project

EmilySalievski reviewed (approved):
Looks good and is working for me

YueYaoYan commented on server/app/main.py:
```diff
@@ -55,6 +67,17 @@ async def lifespan(app: FastAPI):
 
+
+
+class ProjectCreate(BaseModel):
```
I think you can rename ProjectCreate and TranscriptionCreate to Project and Transcription and have them moved to `server/models` (create a new folder). 

 project and transcription related helper functions like `_project_row_to_dict` can also be moved to `server/helpers`.

`main.py` is getting cluttered and I think we should use this for API endpoints. :>

rohith-chaps88 committed:
made helper and model folder for helper functions for projects and transcriptions.

Gunnraj-D commented on server/app/main.py:
```diff
@@ -44,6 +49,13 @@ async def lifespan(app: FastAPI):
 # --- Middleware ---
 
+# new
+DB_PATH = str((Path(__file__).resolve().parent / "qualAI.db"))
```
DB path should also be defined within config.py, then imported in

Gunnraj-D committed:
FIX:
- Moved api models to dedicated file, which are now used in main.py
- Moved DB filepath into config, changed main.py to compluy with this
- Moved project and transcription database references into application lifespan startup

Gunnraj-D committed:
FIX: Fixed import errors

Gunnraj-D committed:
FEAT:
- AI Chat history is now distinct for each project
- AI now only utilsies data for the currenlty selected project

Gunnraj-D committed:
FIX: added dummy project

Gunnraj-D committed:
Merge branch 'develop' into rohith-database_multiple_transactions

Gunnraj-D committed:
FIX: Refactored projects endpoint into api.jsx

Amaan17 reviewed (approved):
Multiple projects can successfully be created and the chats run on different sessions locally. Nice!

Gunnraj-D reviewed (approved):
LGTM
